checkov/terraform/plan_parser.py

```diff
@@ -238,14 +239,24 @@ def _get_resource_changes(template: dict[str, Any]) -> dict[str, dict[str, Any]]
     """Returns a resource address to resource changes dict"""
 
     resource_changes_map = {}
-
     resource_changes = template.get("resource_changes")
-    if resource_changes and isinstance(resource_changes, list):
-        resource_changes_map = {
-            change.get("address", ""): change
-            for change in resource_changes
-        }
 
+    if resource_changes and isinstance(resource_changes, list):
+        for each in resource_changes:
+            resource_changes_map[each["address"]] = each
+            changes = []
+
+            # before + after are None when resources are created/destroyed, so make them safe
+            if not each["change"]["before"]:
+                each["change"]["before"] = {}
+            if not each["change"]["after"]:
+                each["change"]["after"] = {}
+
+            for field in each["change"]["before"]:
+                if each["change"]["before"][field] != each["change"]["after"].get(field):
+                    changes.append(field)
+ 
+            resource_changes_map[each["address"]]["changed_keys"] = changes
```

Comment:
since this is artificially added please use the field name you created, so it won't likely overlap with a real field.
```suggestion
            resource_changes_map[each["address"]][TF_PLAN_RESOURCE_CHANGE_KEYS] = changes
```